### DISCUSSION TITLE
Does PyG offer an official dense-equivalent to its `DataLoader` class?

### DISCUSSION THREAD

**Discussion by amorehead** [pyg-team/pytorch_geometric#8516] [Q&A]
Hello.

I was curious, does PyG offer an official dense-equivalent to its `DataLoader` class? In particular, does PyG offer a collate function that can take a list of `Data` or `HeteroData` objects and create a `Batch` object from them? This `Batch` object, instead of sparsely concatenating all graph features, would concatenate them along a new `batch_size` first dimension for the corresponding feature tensor. Moreover, each (sub)graph may have a different number of nodes, so `DenseDataLoader` wouldn't seem to work for this use case.

**Comment by rusty1s** [comment]
We provide support for batching along a new dimension via https://pytorch-geometric.readthedocs.io/en/latest/advanced/batching.html#batching-along-new-dimensions. Does this work for your use-case? However, note that we don't pad features by default.

**Reply by amorehead** [reply]
Hello. Thank you for pointing this out. I did try this approach, but it doesn't seem to work when you need to pad features (when they have different numbers of elements).

**Reply by amorehead** [reply]
This new PR seems to fix the issue: https://github.com/pyg-team/pytorch_geometric/pull/8518.
